spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -777,8 +777,21 @@ public synchronized void stop() {
 	/**
 	 * Perform a rollback, handling rollback exceptions properly.
 	 * @param ex the thrown application exception or error
+	 * @deprecated in favor of {@link #rollbackOnExceptionIfNecessary(Throwable, long)}
```

Comment:
Why deprecate? Looks like you really have a heavy logic based on the `tag < 0`. So, a dedicated API with a default `-1` for `tag` doesn't look bad....